### PR TITLE
[processing] Add algorithm to extract min/max pixel from raster

### PR DESCRIPTION
**Pull request by nyalldawson**
This algorithm extracts extrema (minimum and maximum) values from a given band of the raster layer.

The output is a vector layer containing point features for the selected extrema, at the center of the associated pixel.

If multiple pixels in the raster share the minimum or maximum value, then only one of these pixels will be included in the output.

The algorithm uses raster iterator to remain efficient on huge rasters, and does not require reading the entire raster to memory

### CODE REVIEW

File: src/analysis/processing/qgsalgorithmrasterminmax.cpp [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+      throw QgsProcessingException( invalidSinkError( parameters, QStringLiteral( "OUTPUT" ) ) );
+  }
+
+  const int extractType = parameterAsInt( parameters, QStringLiteral( "EXTRACT" ), context );
+
+  const int maxWidth = QgsRasterIterator::DEFAULT_MAXIMUM_TILE_WIDTH;
+  const int maxHeight = QgsRasterIterator::DEFAULT_MAXIMUM_TILE_HEIGHT;
+  const int numberBlocksWidth = static_cast< int >( std::ceil( 1.0 * mLayerWidth / maxWidth ) );
+  const int numberBlocksHeight = static_cast< int >( std::ceil( 1.0 * mLayerHeight / maxHeight ) );
+  const int numberBlocks = numberBlocksWidth * numberBlocksHeight;
```
Comment (rouault) [inline src/analysis/processing/qgsalgorithmrasterminmax.cpp:127]: highly unlikely to overflow, but who knows...
```suggestion
  const qint64 numberBlocks = static_cast<qint64>(numberBlocksWidth) * numberBlocksHeight;
```

File: src/analysis/processing/qgsalgorithmrasterminmax.cpp [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+  auto blockRowColToXY = [this]( const QgsRectangle & blockExtent, int row, int col ) -> QgsPointXY
+  {
+    return QgsPointXY( blockExtent.xMinimum() + mRasterUnitsPerPixelX * ( col + 0.5 ),
+                       blockExtent.yMaximum() - mRasterUnitsPerPixelY * ( row + 0.5 ) );
+  };
+
+  QgsRectangle blockExtent;
+  while ( iter.readNextRasterPart( mBand, iterCols, iterRows, rasterBlock, iterLeft, iterTop, &blockExtent ) )
+  {
+    feedback->setProgress( 100 * ( ( iterTop / static_cast< double >( maxHeight * numberBlocksWidth ) ) + iterLeft / static_cast< double >( maxWidth ) ) / numberBlocks );
```
Comment (rouault) [inline src/analysis/processing/qgsalgorithmrasterminmax.cpp:152]: just a thought: would be nice if the QgsRasterIterator could provide that progress percentage...

File: src/analysis/processing/qgsalgorithmrasterminmax.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/analysis/processing/qgsalgorithmrasterminmax.cpp:127]: I wonder if a raster that big exists anywhere :laughing:

Comment (rouault) [reply src/analysis/processing/qgsalgorithmrasterminmax.cpp:127]: > I wonder if a raster that big exists anywhere 😆

a real one, perhaps not, but you could easily create an hostile very small VRT with huge dimensions... (I've fixed so many integer overflows while fixing hundreds of ossfuzz GDAL bugs that pretty much any integer addition and multiplication has become suspicious to my eyes)

Comment (rouault) [reply src/analysis/processing/qgsalgorithmrasterminmax.cpp:127]: Never challenge me for that kind of game ;-) Here's such a GeoPackage: [huge.gpkg.zip](https://github.com/user-attachments/files/17625676/huge.gpkg.zip) : 100 million pixel x 100 million pixel in 98 KB

File: src/analysis/processing/qgsalgorithmrasterminmax.cpp (hunk 2 above)
Comment (nyalldawson) [reply src/analysis/processing/qgsalgorithmrasterminmax.cpp:152]: Ok, done!

File: src/analysis/processing/qgsalgorithmrasterminmax.cpp [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+
+QgsRasterMinMaxAlgorithm *QgsRasterMinMaxAlgorithm::createInstance() const
+{
+  return new QgsRasterMinMaxAlgorithm();
+}
+
+bool QgsRasterMinMaxAlgorithm::prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback * )
+{
+  QgsRasterLayer *layer = parameterAsRasterLayer( parameters, QStringLiteral( "INPUT" ), context );
+  const int band = parameterAsInt( parameters, QStringLiteral( "BAND" ), context );
```
Comment (alexbruy) [inline src/analysis/processing/qgsalgorithmrasterminmax.cpp:86]: Do we need two variables with the band number here?

Comment (nyalldawson) [reply src/analysis/processing/qgsalgorithmrasterminmax.cpp]: Oops, fixed!

**Review by rouault — APPROVED** [review]

**Pull request merged**